jkeohan/fe-mentor-job-listings
# [Refactor] added context and reducer

### What is this PR solving?
Resolves: https://github.com/jkeohan/fe-mentor-job-listings/issues/8
Resolves: https://github.com/jkeohan/fe-mentor-job-listings/issues/25


### Screenshots/Video


### Focus for reviewers
Adding context and reducers only moved the logic out of App but didn't change any of the functionality so just make sure the app works as normal.  

### Testing instructions
[Deploy Preview Link](https://deploy-preview-22--fe-mentor-job-listings.netlify.app/)

1. Click on a tag and confirm that search filter activates to display the tag and only those joblistings that contain that tag are rendered
2. Click on the same tag and confirm the search filter disappears and all joblistings render

### Confirm 

 [x] I tested and confirmed the changes
 [x] This PR is safe to rollback.
 [x] Added the correct label
 [x] This PR follows the recommendations regarding squashig commits
